docs/source/deployment-considerations.rst

```diff
@@ -0,0 +1,91 @@
+
+Especially when used by teams, mature Dask deployments will want to avoid users passing their own credentials to the cluster directly in code, using strategies such as generating temporary tokens for users, authenticating workers under service accounts, etc.
+
+
+Controlling access
+------------------
+Most non-commercial deployment libraries rely on the user launching the cluster to have access to the underlying system the cluster will run on (such as a cloud provider, an HPC cluster, Kubernetes, etc.). When enabling a team to use Dask, this may not be the case: you want to let users launch clusters on demand, which might create cloud VMs or Kubernetes pods, without actually giving them permission to create cloud VMs or Kubernetes pods directly. `Dask-gateway <https://gateway.dask.org/>`_ is a common way to do this, but that does require additional administration.
+
+Most Dask deployment options set reasonable defaults for access (i.e. not making your cluster accessible to anyone on the Internet), but you still should make sure that your clusters (or your users' clusters) aren't accessible to unauthorized users.
+
+Additionally, if you're connecting to a cluster over the Internet, you should ensure that that connection is encrypted, since sensitive information, such as credentials or proprietary data, may flow over it. You might do this by port-forwarding your connection over SSH, using :doc:`TLS <tls>`, or using `Dask-gateway <https://gateway.dask.org/>`_ or a commercial offering that manages this automatically.
+
+
+Controlling cost
+----------------
+It's easy to forget to shut down a cluster and run up an expensive bill over the weekend. Some deployment libraries also may not always be able to fully clean up a cluster—for example, `dask-cloudprovider <https://cloudprovider.dask.org/>`_ won't clean up cloud resources if the client Python process (or machine!) shuts down abruptly.
+
+Therefore, when launching clusters automatically in production, or enabling many team members to launch them, you should be confident that all resources will be cleaned up, or shut down if they exceed a cost threshold.
+
+When managing Dask for a team, you may also want a way to limit how much individual users can spend, to prevent accidental overruns.
+
+
+Monitoring cost
+---------------
+It's good to be able to answer questions such as:
+
+- How much are we spending on Dask?
+- What are we spending it on? (machines, machines that should have been turned off, network egress that shouldn't have happened, etc.)
+- Who/what is responsible?
+
+Non-commercial deployment tools generally don't build in this sort of monitoring. Organizations that need it either end up building their own tools, or turning to commercial deployment offerings.
```

Comment:
Maybe soften this a bit.

```suggestion
Many deployment tools generally don't build in this sort of monitoring. Organizations that need it either end up building their own tools, or turning to commercial deployment offerings.
```

Reply:
I'd originally written that, but then couldn't think of any non-commercial tools that actually did have built-in capabilities for cost monitoring. Is there something I'm not thinking of?

Arguably Coiled doesn't even do what I've described here. Coiled can tell you how you spent your _Coiled_ bill, but for your AWS bill, you still have to look yourself in the cost explorer (though this is facilitated by tags Coiled adds to all your dask infrastructure).